bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TypedescValue.java

```diff
@@ -44,7 +44,7 @@
 
     @Deprecated
     public TypedescValue(BType describingType) {
-        this.type = BTypes.typeTypedesc;
+        this.type = new BTypedescType(describingType);
         this.describingType = describingType;
```

Comment:
Do we need the `describingType` field anymore, since that is already there in the type now?

Reply:
On a second thought, I think we may have to keep it. Please ignore my previous comment :)

Reply:
OK :)